File: examples/Map.js
#!/usr/bin/env node

/* jshint
    esversion: 6,
    evil:      true
*/

// ------
// Map.js
// ------

// https://www.w3schools.com/jsref/jsref_map.asp

"use strict";

const assert = require('assert');
const _      = require('lodash');

function test1 () {
    const m = _.map([2, 3, 4], (x) => {return x * x;});
    assert((typeof m) === "object");
    assert(m instanceof Array);
    assert(m instanceof Object);
    assert(_.isEqual(m, [4, 9, 16]));
    assert(_.isEqual(m, [4, 9, 16]));}

function test2 () {
    const m = _.map([], null);
    assert((typeof m) === "object");
    assert(m instanceof Array);
    assert(m instanceof Object);
    assert(_.isEqual(m, []));}

function test3 () {
    const m = [2, 3, 4].map((x) => {return x * x;});
    assert((typeof m) === "object");
    assert(m instanceof Array);
    assert(m instanceof Object);
    assert(_.isEqual(m, [4, 9, 16]));
    assert(_.isEqual(m, [4, 9, 16]));}

function main () {
    for (let i of _.range(3))
        eval("test" + (i + 1) + "()");}

console.log("Map.js");
main();
console.log("Done.");

File: examples/Map.py
#!/usr/bin/env python3

# pylint: disable = bad-whitespace
# pylint: disable = eval-used
# pylint: disable = invalid-name
# pylint: disable = missing-docstring

# -------
# Map1.py
# -------

# https://docs.python.org/3/library/functions.html#map

def test1 () :
    m = map(lambda x : x ** 2, [2, 3, 4])
    assert hasattr(m, "__next__")
    assert hasattr(m, "__iter__")
    assert iter(m) is m
    assert list(m) == [4, 9, 16]
    assert list(m) == []

def test2 () :
    m = map(lambda x : x ** 2, {2, 3, 4})
    assert set(m) == {4, 9, 16}
    assert set(m) == set()

def test3 () :
    m = map(None, ())
    assert list(m) == []

def main () :
    for n in range(3) :
        eval("test" + str(n + 1) + "()")

if __name__ == "__main__" : # pragma: no cover
    print("Map.py")
    main()
    print("Done.")
